pkg/argocd/update_test.go

```diff
@@ -3643,3 +3643,58 @@ func Test_GetRepositoryLock(t *testing.T) {
 	require.NotNil(t, state.repositoryLocks[repo2])
 	require.Equal(t, lock3, state.repositoryLocks[repo2])
 }
+
+func Test_mergeKustomizeOverride(t *testing.T) {
```

Comment:
I was trying to test the new function from a higher level, at calling `marshalParamsOverride` instead.  Do you think it is worthwhile to do it from that call?  eg. see the first couple of scenarios from the existing test `Test_MarshalParamsOverride`

Reply:
I can add a sub-test to `Test_MarshalParamsOverride` by setting the new images in app spec, to test out the changes from that level.

Reply:
Ok, I see the call to marshalParamsOverride